src/setups/complex_shape.jl

```diff
@@ -0,0 +1,108 @@
+                                                                winding_number_factor=sqrt(eps())),
+                 seed=shape.min_corner, max_nparticles=Int(1e7),
+                 pad_initial_particle_grid=2particle_spacing)
+
+Complex shape filled with particles. Returns an [`InitialCondition`](@ref).
+Note that an initial particle grid is generated inside the bounding box of the shape.
+An `point_in_shape_algorithm` checks if particles are inside the shape or not.
+For more information about the method see [`WindingNumberJacobson`](@ref) or [`WindingNumberHorman`](@ref).
+
+# Arguments
+- `shape`: Complex shape returned by [`load_shape`](@ref).
+
+# Keywords
+- `particle_spacing`:   Spacing between the particles.
+- `density`:            Either a function mapping each particle's coordinates to its density,
+                        or a scalar for a constant density over all particles.
+- `velocity`:           Either a function mapping each particle's coordinates to its velocity,
+                        or, for a constant fluid velocity, a vector holding this velocity.
+                        Velocity is constant zero by default.
+- `mass`:               Either `nothing` (default) to automatically compute particle mass from particle
+                        density and spacing, or a function mapping each particle's coordinates to its mass,
+                        or a scalar for a constant mass over all particles.
+- `pressure`:           Scalar to set the pressure of all particles to this value.
+                        This is only used by the [`EntropicallyDampedSPHSystem`](@ref) and
+                        will be overwritten when using an initial pressure function in the system.
+                        Cannot be used together with hydrostatic pressure gradient.
+- `point_in_shape_algorithm`: Algorithm for sampling the complex shape with particles.
+                              It basically checks whether a particle is inside an object or not.
+                              For more information see [`WindingNumberJacobson`](@ref) or [`WindingNumberHorman`](@ref)
+- `seed`:               Coordinates of the minimum corner of the initial particle grid (excluding padding).
+- `max_nparticles`:     Maximum number of particles in the initial particle grid.
+- `pad_initial_particle_grid`: Padding of the initial particle grid.
+
+
+!!! warning "Experimental Implementation"
+    This is an experimental feature and may change in any future releases.
+"""
+function ComplexShape(shape::Shapes; particle_spacing, density, pressure=0.0, mass=nothing,
+                      velocity=zeros(ndims(shape)),
+                      point_in_shape_algorithm=WindingNumberJacobson(; shape,
+                                                                     hierarchical_winding=false,
+                                                                     winding_number_factor=sqrt(eps())),
```

Comment:
This factor is never used?

Reply:
Yes, it is. In the `point_in_poly` function to relax the winding number condition

Reply:
Ah, GitHub formatting made it look like it was a kwarg to this function here.